snapshot.go

```diff
@@ -458,7 +464,7 @@ func WriteSnapshot(ctx context.Context, _ uint64, db *DB, w io.Writer, offline b
 
 					// If file compaction is enabled we do not need to snapshot the parquet parts.
 					for _, cfg := range db.columnStore.indexConfig {
-						if cfg.Type == CompactionTypeParquetDisk {
+						if cfg.Type == index.CompactionTypeParquetDisk {
 							return ErrSkipPart
```

Comment:
I feel like we're treating snapshots and index snapshots as two separate things (i.e. the index `Snapshot` method and then this `WriteSnapshot` that skips parts) which is confusing to me from a first glance. Can we unify these two snapshot paths?

Reply:
I think I would like to refactor this to have the snapshot function just call the `index.Snapshot(...)` and have it correctly snapshot itself. But I think that this PR is already a bit too large as is. So I'm going to defer a change like that.

Reply:
Sounds good, yes I think delegating the snapshotting to the index sounds like the solution here (i.e. removing part iteration from this file).